File: frontend/src/core/query-objects/FiltroPadraoQuery.ts
export default interface FiltroPadraoQuery {
    busca?: string;
    pagina?: number;
    tamanho?: number;
    de?: Date;
    ate?: Date;
    ordenarPor?: string;
    direcao?: 'asc' | 'desc';
    status?: string;
    idRepresentante?: string;
    idGerente?: string;
    idCliente?: string;
}

File: backend/API/Models/Requests/FiltroPadraoQuery.cs
namespace RedMobilePedidos.API.Models.Requests;

public sealed record FiltroPadraoQuery
{
    public DateTime? De { get; init; }
    public DateTime? Ate { get; init; }
    public int? Pagina { get; init; }
    public int? Tamanho { get; init; }
    public string? OrdenarPor { get; init; }
    public string? Direcao { get; init; }
    public string? Busca { get; init; }
    public string? Status { get; init; }
    public string? Cnpj { get; init; }
    public string? FiltroUsuario { get; init; }
    public string? IdGerente { get; init; }
    public string? IdRepresentante { get; init; }
    public string? IdCliente { get; init; }
}
